src/fdc/MSXtar.cc

```diff
@@ -193,38 +193,45 @@ MSXtar::~MSXtar()
 	}
 }
 
-// transform BAD_FAT (0xFF7) and EOF_FAT-range (0xFF8-0xFFF)
-// to a single value: EOF_FAT (0xFFF)
-static constexpr unsigned normalizeFAT(unsigned cluster)
-{
-	return (cluster < BAD_FAT) ? cluster : EOF_FAT;
-}
-
 // Get the next cluster number from the FAT chain
-unsigned MSXtar::readFAT(unsigned clNr) const
+unsigned MSXtar::readFAT(unsigned index) const
 {
 	assert(!fatBuffer.empty()); // FAT must already be cached
+	assert(index >= FIRST_CLUSTER && index < FIRST_CLUSTER + clusterCount);
```

Comment:
This subexpression: `(FIRST_CLUSTER <= index) && (index < (FIRST_CLUSTER + clusterCount))`
is repeated 5x in this commit (maybe more if I look at the full code?)
Maybe factor it out in a `isValidClusterIndex()` helper function? (Or `clusterIndexInRange()`?)

Reply:
Indeed, it’s factored out later in the Cluster class commit.